src/computerAlgorithms.js

```diff
@@ -0,0 +1,202 @@
+const lines = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+];
+const diagonalLines = lines.slice(6);
+const straightLines = lines.slice(0,6);
+
+function firstStepAlgorithm(squares, playerGoesFirst){
+  if (playerGoesFirst){
+    if (!squares[4]) {
+      return 4;
+    } else {
+      return getRandomAnglePosition(squares);
+    }
+  } else {
+    const firstStepInCenter = Math.round(Math.random());
+
+    if (firstStepInCenter) {
+      return 4;
+    } else {
+      return getRandomAnglePosition(squares);
+    }
+  }
```

Comment:
```suggestion
const goodFirstStepPosition = 4; // it's just an example and you shouldn't use the same name if you don't like it

function calculateFirstStep(squares, isPlayerFirst){
  if (isPlayerFirst) {
   const firstStepInCenter = Math.round(Math.random());
   return !squares[goodFirstStepPosition] || firstStepInCenter ? goodFirstStepPosition : 
   getRandomAnglePosition(squares);
 }
}
```
It can be simplified, so try to write less code. Because I see a lot of code, but most of it can be reduced at least in a twice. Just give it a short. It's difficult from beginning, but you should try.

Reply:
Thanks, I made some changes in code to make it shorter